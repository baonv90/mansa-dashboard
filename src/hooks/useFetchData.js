import { useState, useEffect } from 'react';

// A hook to fetch data from given url
// it returns the fetching status loading and data
export const useFetchData = (url) => {
  const [data, setData] = useState();
  const [loading, setLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    setLoading(true);
    fetch(url).then((res) => {
      res.json().then((data) => {
        setData(data);
        setLoading(false);
      })
      .catch(() => {
        setHasError(true);
        setLoading(false);
      });
    });
  }, [url]);

  return { data, loading, hasError };
};