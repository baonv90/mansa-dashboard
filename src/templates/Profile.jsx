import React from 'react';
import styled from 'styled-components';
import get from 'lodash/get';


import { useFetchData } from '../hooks/useFetchData';
const data_source = 'https://randomuser.me/api/';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Picture = styled.img`
  width: 100px;
  height: 100px;
  border-radius: 50%;
  margin-bottom: 1rem;
  border: 3px solid var(--border-color);
`;

export const TitleName = styled.span`
  font-size: 1.1rem;
  font-weight: 600;
  line-height: 1.5rem;
  margin-bottom: .5rem;
`;

export const Info = styled.span`
  font-size: .9rem;
  font-weight: 400;
  line-height: 1.25rem;
`;

// User profile from randomuser
const Profile = () => {
  const { data } = useFetchData(data_source);
  const userInfo = get(data, 'results[0]');

  const renderProfile = ({ phone, email, name, picture } = {}) => {
    const { first, last } = name;
    const displayName = first + ' ' + last;
    return <>
      <Picture src={picture ? picture.medium : ''} />
      <TitleName>{displayName}</TitleName>
      <Info>{email}</Info>
      <Info>{phone}</Info>
    </>
  };

  if (!userInfo) {
    return null;
  }

  return (
    <Container>
      {renderProfile(userInfo)}
    </Container>
  )
};

export default Profile;