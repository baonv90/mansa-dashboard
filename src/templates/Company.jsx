import React from 'react';
import styled from 'styled-components';
import get from 'lodash/get';

import { useFetchData } from '../hooks/useFetchData';
import { TitleName, Info } from './Profile';

const data_source = 'https://entreprise.data.gouv.fr/api/sirene/v3/unites_legales/852379890';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  padding: .5rem;
`;

// Company information from entreprise.data.gouv
const Company = () => {
  const { data } = useFetchData(data_source);
  const companyInfo = get(data, 'unite_legale');

  const renderInfo = (companyInfo) => {
    const {
      denomination,
      etablissement_siege: {
        date_creation,
        siret,
        geo_adresse
      } 
    } = companyInfo || {};
  
    return <>
      <TitleName>{denomination}</TitleName>
      <Info>Siret: {siret}</Info>
      <Info>Date Creation: {date_creation}</Info>
      <Info>Adresse: {geo_adresse}</Info>
    </>;
  };

  if (!companyInfo) {
    return null;
  }
  
  return (
    <Container>
      {renderInfo(companyInfo)}
    </Container>
  )
};

export default Company;