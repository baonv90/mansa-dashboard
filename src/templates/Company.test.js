import renderer from 'react-test-renderer';
import Company from './Company';
import { TitleName } from './Profile';
import { useFetchData } from '../hooks/useFetchData';

jest.mock('../hooks/useFetchData');

describe('templates > Company', () => {
  const fakeData = {
    unite_legale: {
      denomination: 'Mansa',
      etablissement_siege: {
        date_creation: '01/01/2000',
        siret: '123457654765',
        geo_adresse: '1 Avenue de Paris, Paris, 75001'
      } 
    }
  };

  beforeEach(() => {
    useFetchData.mockReturnValue({
      loading: false,
      data: fakeData
    });
  });

  it('should match snapshot', async () =>  {
    const component = await renderer.create(<Company />);
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should pass data correctly', async () =>  {
    const tree = await renderer.create(<Company />);
    
    const title = tree.root.findByType(TitleName);
    expect(title.props.children).toEqual('Mansa');    
  });
});
