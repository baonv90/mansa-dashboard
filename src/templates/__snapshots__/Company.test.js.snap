// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates > Company should match snapshot 1`] = `
<div
  className="sc-eCApnc wGQyr"
>
  <span
    className="sc-dlnjwi emJSdu"
  >
    Mansa
  </span>
  <span
    className="sc-hKFxyN hpHuXd"
  >
    Siret: 
    123457654765
  </span>
  <span
    className="sc-hKFxyN hpHuXd"
  >
    Date Creation: 
    01/01/2000
  </span>
  <span
    className="sc-hKFxyN hpHuXd"
  >
    Adresse: 
    1 Avenue de Paris, Paris, 75001
  </span>
</div>
`;
